### PR TITLE
Reconcile vSphere Tags and Tag Categories 

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
In a previous iteration, a new reconciliation framework has been introduced for the vSphere cloud provider. This PR reconcile all Tags and Tag Categories that are attached to any vSphere user cluster, where we stopped creating those tag categories unconditionally and we demand that, this category is created before hand and passed on to the user cluster.

vSphere cloud provider will make sure that the category exists before moving forward with the reconciliation and make sure that whatever tag is added to this cluster, is gonna be created in that category if it doesn't exist and upon cluster updates or deletion, we make sure that, those delete tags are synced and only delete tags that are attached on the cluster level.    
**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**
/kind cleanup
/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Cleanup default tag categories creation and only create tags in the tag category which assigned on the user cluster level
```

**Documentation**:

```documentation
None
```

### CODE REVIEW

**Review by xrstf — APPROVED** [review]
/approve

File: pkg/provider/cloud/vsphere/provider.go [hunk 1]
```diff
@@ -34,10 +33,7 @@ import (
 const (
 	folderCleanupFinalizer = "kubermatic.k8c.io/cleanup-vsphere-folder"
 	// categoryCleanupFinilizer will instruct the deletion of the default category tag.
-	tagCategoryCleanupFinilizer = "kubermatic.k8c.io/cleanup-vsphere-tag-category"
-
-	defaultCategoryPrefix = "kubermatic.k8c.io/tag-category"
-	defaultTagPrefix      = "kubermatic.k8c.io/tag"
+	tagCleanupFinilizer = "kubermatic.k8c.io/cleanup-vsphere-tags"
```
Comment (ahmedwaleedmalik) [inline pkg/provider/cloud/vsphere/provider.go]: s/tagCleanupFinilizer/tagCleanupFinalizer

**Review by ahmedwaleedmalik — CHANGES_REQUESTED** [review]

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve

**Comment by embik** [comment]
@moadqassem it seems that this PR changed provider behaviour (i.e., categories need to be created beforehand). Shouldn't this require a documentation change then?

**Comment by embik** [comment]
It's also unclear if the API changes in this PR require a migration of existing clusters. @WeirdMachine and I are not sure about that.

**Pull request merged**